lib/cupsffi/printer.rb

```diff
@@ -165,13 +165,19 @@ def print_data(data, mime_type, options = {})
       raise last_error
     end
 
+    # Copy the data into a char pointer to allow null bytes
+    data_pointer = FFI::MemoryPointer.new(:char, data.bytesize)
+    data_pointer.put_bytes(0, data)
+
     http_status = CupsFFI::cupsStartDocument(@connection, @name,
                                              job_id, 'my doc', mime_type, 1)
 
-    http_status = CupsFFI::cupsWriteRequestData(@connection, data, data.length)
+    http_status = CupsFFI::cupsWriteRequestData(@connection, data_pointer, data.length)
```

Comment:
Could you change `data.length` to `data.bytesize` while you're in here?  The C code is expecting byte length for this 3rd parameter, not the string length.

https://github.com/apple/cups/blob/57b7b66b58a66426494ec13ffb18f730afeab8b5/cups/request.c#L895

I realize this is my own bug, but if you could change that at the same time I'd appreciate it.